src/app/core/services/transaction.service.ts

```
@@ -845,12 +850,31 @@ export class TransactionService {
+        // For advance wallet accounts, ensure source_account_id is null
+        if (account.type === 'PERSONAL_ADVANCE_ACCOUNT') {
+          return of({
+            source_account_id: null,
+            skip_reimbursement: true,
+            advance_wallet_id: account.id,
+          });
+        }
+
+        return of({
+          source_account_id: account.id,
+          skip_reimbursement: account.type === 'PERSONAL_CASH_ACCOUNT' ? false : true,
```

Biome (1.9.4)

[error] Unnecessary use of boolean literals in conditional expression.

Simplify your code by directly assigning the result without using a ternary operator.
If your goal is negation, you may use the logical NOT (!) or double NOT (!!) operator for clearer and concise code.
 Check for more details about NOT operator.
Unsafe fix: Remove the conditional expression with


(lint/complexity/noUselessTernary)